components/os/unix.go

```diff
@@ -48,34 +46,40 @@ func getDefaultInstanceType(env config.Environment, arch Architecture) string {
 	}
 }
 
-func getUnixInstallFormatString(scriptName string, version AgentVersion) string {
-	if version.PipelineID != "" {
-		testEnvVars := []string{}
-		testEnvVars = append(testEnvVars, "TESTING_APT_URL=apttesting.datad0g.com")
-		// apt testing repo
-		// TESTING_APT_REPO_VERSION="pipeline-xxxxx-a7 7"
-		testEnvVars = append(testEnvVars, fmt.Sprintf(`TESTING_APT_REPO_VERSION="%v-a7 7"`, version.PipelineID))
-		testEnvVars = append(testEnvVars, "TESTING_YUM_URL=yumtesting.datad0g.com")
-		// yum testing repo
-		// TESTING_YUM_VERSION_PATH="testing/pipeline-xxxxx-a7/7"
-		testEnvVars = append(testEnvVars, fmt.Sprintf("TESTING_YUM_VERSION_PATH=testing/%v-a7/7", version.PipelineID))
-		commandLine := strings.Join(testEnvVars, " ")
-
-		return fmt.Sprintf(
-			`DD_API_KEY=%%s %v bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/%v)"`,
-			commandLine,
-			scriptName)
+func getUnixRepositoryParams(version AgentVersion) string {
+	envVars := []string{}
+	switch version.Repository {
+	case TrialRepository, TestingRepository:
+		aptChannel := version.Channel
+		yumChannel := version.Channel
+		if version.Repository == TestingRepository {
+			aptChannel = Channel(fmt.Sprintf("pipeline-%v-a%v", version.PipelineID, version.Major))
+			yumChannel = Channel(fmt.Sprintf("testing/pipeline-%v-a%v", version.PipelineID, version.Major))
+		}
+
+		envVars = append(envVars, fmt.Sprintf(`TESTING_APT_URL="apt%v.datad0g.com"`, version.Repository))
+		envVars = append(envVars, fmt.Sprintf(`TESTING_APT_REPO_VERSION="%v %v"`, aptChannel, version.Major))
+		envVars = append(envVars, fmt.Sprintf(`TESTING_YUM_URL="yum%v.datad0g.com"`, version.Repository))
+		envVars = append(envVars, fmt.Sprintf(`TESTING_YUM_VERSION_PATH="%v/%v"`, yumChannel, version.Major))
+	case StagingRepository:
+		envVars = append(envVars, `DD_REPO_URL="datad0g.com"`)
+		envVars = append(envVars, fmt.Sprintf(`DD_AGENT_DIST_CHANNEL="%v"`, version.Channel))
+	case ProdRepository:
+		envVars = append(envVars, `DD_REPO_URL="datadoghq.com"`)
+		envVars = append(envVars, fmt.Sprintf(`DD_AGENT_DIST_CHANNEL="%v"`, version.Channel))
 	}
 
+	return strings.Join(envVars, " ")
+}
+
+func getUnixInstallFormatString(scriptName string, version AgentVersion) string {
 	commandLine := fmt.Sprintf("DD_AGENT_MAJOR_VERSION=%v ", version.Major)
 
 	if version.Minor != "" {
 		commandLine += fmt.Sprintf("DD_AGENT_MINOR_VERSION=%v ", version.Minor)
 	}
 
-	if version.BetaChannel {
-		commandLine += "REPO_URL=datad0g.com DD_AGENT_DIST_CHANNEL=beta "
-	}
+	commandLine += getUnixRepositoryParams(version)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Not a blocker: rather than a command line, this is a comma separated list of env variables